BizTravel360/src/main/java/com/travel/BizTravel360/accommodation/AccommodationService.java

```diff
@@ -1,10 +1,133 @@
 package com.travel.BizTravel360.accommodation;
 
-import java.util.List;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.travel.BizTravel360.accommodation.exeptions.AccommodationNotFoundException;
+import com.travel.BizTravel360.accommodation.exeptions.AccommodationSaveException;
+import com.travel.BizTravel360.file.FileService;
+import jakarta.transaction.Transactional;
+import jakarta.validation.*;
+import org.hibernate.validator.messageinterpolation.ParameterMessageInterpolator;
+import org.springframework.beans.factory.annotation.Value;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
 
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.*;
+
+@Slf4j
+@Transactional
+@Service
 public class AccommodationService implements AccommodationRepository {
+    
+    private final FileService fileService;
+    @Value("${accommodations.file.path}")
+    private String accommodationFilePath;
+    
+    private List<Accommodation> accommodations = new ArrayList<>();
+    
+    public AccommodationService(FileService fileService, @Value("${accommodations.file.path}") String accommodationFilePath) {
+        this.fileService = fileService;
+        this.accommodationFilePath = accommodationFilePath;
+    }
+    
+    @Override
+    public void saveAccommodation(Accommodation accommodation) throws IOException {
+        try {
+            trimAccommodation(accommodation);
+            validateAccommodation(accommodation);
+            
+            accommodation.setAccommodationId(UUID.randomUUID().getMostSignificantBits() & Long.MAX_VALUE);
+            List<Accommodation> existingAccommodations = fetchAccommodationList();
+            existingAccommodations.add(accommodation);
+            
+            fileService.writerToFile(existingAccommodations, accommodationFilePath);
+            
+            logSuccessMessage(accommodation);
+        } catch (IOException e) {
+            log.error("Failed to save accommodation {}", accommodation, e);
+            throw new AccommodationSaveException(String.format("Failed to save accommodation %s.", accommodation), e);
+        }
+    }
+    
     @Override
-    public List<Accommodation> findAll() {
-        return List.of();
+    public List<Accommodation> fetchAccommodationList() throws IOException {
+        if (Files.exists(Paths.get(accommodationFilePath))){
+            this.accommodations = fileService.readFromFile(accommodationFilePath, new TypeReference<List<Accommodation>>() {});
```

Comment:
```suggestion
            List<Accomodation> accomodationList = accommodations = fileService.readFromFile(accommodationFilePath, new TypeReference<List<Accommodation>>() {});
```

I tak w każdym obiekcie, transport, employee